bigtable-hbase-2.x-parent/bigtable-hbase-2.x/src/main/java/com/google/cloud/bigtable/hbase2_x/BigtableAdmin.java

```diff
@@ -119,6 +135,35 @@ public void createTable(TableDescriptor desc, byte[] startKey, byte[] endKey, in
     return response;
 
   }
+  
+  @Override
+  public List<SnapshotDescription> listSnapshots()
+      throws IOException {
+    ListSnapshotsRequest request = ListSnapshotsRequest.newBuilder()
+        .setParent(getSnapshotClusterName().toString())
+        .build();
+
+    ListSnapshotsResponse snapshotList = Futures.getChecked(bigtableTableAdminClient
+        .listSnapshotsAsync(request), IOException.class);
+
+    List<SnapshotDescription> response = new ArrayList<>();
+
+    for (Snapshot snapshot : snapshotList.getSnapshotsList()) {
+      BigtableSnapshotName snapshotName = new BigtableSnapshotName(snapshot.getName());
+      BigtableTableName tableName = new BigtableTableName(snapshot.getSourceTable().getName());
+      HBaseProtos.SnapshotDescription snapDesc = HBaseProtos.SnapshotDescription.newBuilder()
```

Comment:
We shouldn't need an `HBaseProtos.SnapshotDescription`here.  Just convert from Bigtable's `Snapshot` to HBase's `SnapshotDescription` directly.

Reply:
response.add(new SnapshotDescription(
          snapshot.getName(), 
          TableName.valueOf(snapshot.getSourceTable().getName())));

Does this suffice? What about specifying the owner, type and creation time?

Reply:
I don't think we need to specify owner or type.  Snapshot has a createdTime.